api/src/mailbox.rs

```diff
@@ -1047,6 +1054,98 @@ pub struct AuthorizeAndStashResp {
 }
 impl Response for AuthorizeAndStashResp {}
 
+#[derive(Debug, Default, PartialEq, Eq)]
+pub struct MboxBuffer {
+    pub data: ArrayVec<u8, { Self::MAX_SIZE }>,
+}
+impl MboxBuffer {
+    const MAX_SIZE: usize = 4096;
+}
+
+impl PartialEq<[u8]> for MboxBuffer {
+    fn eq(&self, other: &[u8]) -> bool {
+        self.data.as_slice() == other
+    }
+}
+
+impl Deref for MboxBuffer {
+    type Target = ArrayVec<u8, { Self::MAX_SIZE }>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.data
+    }
+}
+
+impl DerefMut for MboxBuffer {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.data
+    }
+}
```

Comment:
Should we remove these now that they aren't used?

Reply:
Done